File: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/examples-js/AppServiceEnvironments_ListMultiRolePools.js
const { WebSiteManagementClient } = require("@azure/arm-appservice");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Description for Get all multi-role pools.
 *
 * @summary Description for Get all multi-role pools.
 * x-ms-original-file: specification/web/resource-manager/Microsoft.Web/stable/2022-03-01/examples/AppServiceEnvironments_ListMultiRolePools.json
 */
async function getAllMultiRolePools() {
  const subscriptionId = "34adfa4f-cedf-4dc0-ba29-b6d1a69ab345";
  const resourceGroupName = "test-rg";
  const name = "test-ase";
  const credential = new DefaultAzureCredential();
  const client = new WebSiteManagementClient(credential, subscriptionId);
  const resArray = new Array();
  for await (let item of client.appServiceEnvironments.listMultiRolePools(
    resourceGroupName,
    name
  )) {
    resArray.push(item);
  }
  console.log(resArray);
}

getAllMultiRolePools().catch(console.error);

File: specification/web/resource-manager/Microsoft.Web/AppService/stable/2025-03-01/examples-java/AppServiceEnvironments_ListMultiRolePools.java

/**
 * Samples for AppServiceEnvironments ListMultiRolePools.
 */
public final class Main {
    /*
     * x-ms-original-file: specification/web/resource-manager/Microsoft.Web/AppService/stable/2025-03-01/examples/
     * AppServiceEnvironments_ListMultiRolePools.json
     */
    /**
     * Sample code: Get all multi-role pools.
     * 
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void getAllMultiRolePools(com.azure.resourcemanager.AzureResourceManager azure) {
        azure.webApps().manager().serviceClient().getAppServiceEnvironments().listMultiRolePools("test-rg", "test-ase",
            com.azure.core.util.Context.NONE);
    }
}
